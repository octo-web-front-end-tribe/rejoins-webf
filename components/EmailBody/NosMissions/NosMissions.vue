<template>
    <!-- Thumbnail Left, Text Right : BEGIN -->
    <tr>
        <!-- dir=ltr is where the magic happens. This can be changed to dir=rtl to swap the alignment on wide while maintaining stack order on narrow. -->
        <td dir="ltr" bgcolor="#ffffff" align="center" height="100%" valign="top" width="100%"
            style="padding: 0px 0;">
            <!--[if mso]>
            <table role="presentation" border="0" cellspacing="0" cellpadding="0" align="center" width="780"
                   style="width: 780px;">
                <tr>
                    <td align="center" valign="top" width="780" style="width: 780px;">
            <![endif]-->
            <table role="presentation" border="0" cellpadding="0" cellspacing="0" align="center" width="100%"
                   style="max-width:780px;">
                <tr>
                    <td align="left" valign="top" style="font-size:0; padding: 0px 0;">
                        <!--[if mso]>
                        <table role="presentation" border="0" cellspacing="0" cellpadding="0" align="center" width="780"
                               style="width: 780px;">
                            <tr>
                                <td align="left" valign="top" width="220" style="width: 220px;">
                        <![endif]-->
                        <div
                                style="display:inline-block; margin: 0 -2px; max-width: 200px; min-width:160px; vertical-align:top; width:100%;"
                                class="stack-column">
                            <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
                                <tr>
                                    <td dir="ltr" style="padding: 0 10px 10px 10px;">
                                        <div v-for="(people, index) in peoples" :key="index" style="text-align:center">
                                            <img :src="people.url_images" width="100" height="" border="0"
                                                 alt="alt_text"
                                                 class="center-on-narrow"
                                                 style="width: 100%; max-width: 100px; height: 100px; margin-top: 20px; margin-bottom: 5px; background: #dddddd; font-family: sans-serif; font-size: 15px; line-height: 140%; color: #555555;">
                                            <h2 style="margin: 0; font-family: sans-serif; font-size: 20px; line-height: 125%; color: rgb(20, 160, 192); font-weight: normal;">
                                                {{people.trigramme}}</h2>
                                        </div>
                                    </td>
                                </tr>
                            </table>
                        </div>
                        <!--[if mso]>
                        </td>
                        <td align="left" valign="top" width="440" style="width: 440px;">
                        <![endif]-->
                        <div
                                style="display:inline-block; margin: 0 -2px; max-width:70%; min-width:320px; vertical-align:top;"
                                class="stack-column">
                            <table role="presentation" cellspacing="0" cellpadding="0" border="0" width="100%">
                                <tr>
                                    <td dir="ltr"
                                        style="font-family: sans-serif; font-size: 15px; line-height: 140%; color: #555555; padding: 10px 10px 0; text-align: left;"
                                        class="center-on-narrow">
                                        <h2
                                                style="margin: 0 0 10px 0; font-family: sans-serif; font-size: 24px; line-height: 125%; color: rgb(20, 160, 192); font-weight: bold;">
                                            {{title}}</h2>
                                        <p style="margin: 0 0 10px 0;">
                                            <component :is="component"/>
                                        </p>
                                        <br>
                                    </td>
                                </tr>
                            </table>
                        </div>
                        <!--[if mso]>
                        </td>
                        </tr>
                        </table>
                        <![endif]-->
                    </td>
                </tr>
                <tr>
                    <td>
                        <hr style="width: 60%; color: #dddddd">
                    </td>
                </tr>
            </table>
            <!--[if mso]>
            </td>
            </tr>
            </table>
            <![endif]-->
        </td>
    </tr>
</template>

<script>
    import ANEF from "@/contents/missions/ANEF.md"
    import AVIGNON from "@/contents/missions/AVIGNON.md"
    import conges from "@/contents/missions/conges_fait.md"
    import DISNEY from "@/contents/missions/DISNEY_fait.md"
    import CANAL from "@/contents/missions/CANAL.md"
    import FAURECIA from "@/contents/missions/FAURECIA.md"
    import INVIVO from "@/contents/missions/INVIVO_fait.md"
    import MEETIC from "@/contents/missions/MEETIC_fait.md"
    import PIX from "@/contents/missions/PIX_fait.md"
    import POLICE from "@/contents/missions/POLICE_fait.md"
    import RANDSTAD from "@/contents/missions/RANDSTAD_fait.md"
    import TEXA from "@/contents/missions/TEXA_fait.md"

    export default {
        name: 'nos-missions',
        components: {
            ANEF,
            AVIGNON,
            conges,
            DISNEY,
            CANAL,
            FAURECIA,
            INVIVO,
            MEETIC,
            PIX,
            POLICE,
            RANDSTAD,
            TEXA
        },
        props: ['mission'],

        data() {
            return {
                component: this.mission.texte,
                title: this.mission.name,
                peoples: this.mission.peoples
            }
        }
    };
</script>

<style scoped>

</style>
