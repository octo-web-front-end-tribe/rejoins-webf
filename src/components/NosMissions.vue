<template>
    <div>
        <titre titre="Nos missions"/>
        <markdown-nos-missions/>
    </div>
</template>

<script>
  import Titre from './Titre'
  import MarkdownNosMissions from '@/../contents/nosmissions.md'

  export default {
    name: 'NosMissions',
    components: {Titre, MarkdownNosMissions},
    data () {
      return {}
    },

    created () {
    },

    methods: {}
  }
</script>

<style scoped>

</style>
