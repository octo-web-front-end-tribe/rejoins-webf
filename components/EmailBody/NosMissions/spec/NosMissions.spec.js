import Vue from 'vue'
import NosMissions from '@/components/EmailBody/NosMissions/NosMissions'
import configuration from '@/contents/config'

describe('NosMissions Component', () => {
    let $mounted;

    const Constructor = Vue.extend(NosMissions);

    beforeEach(() => {
        $mounted = new Constructor({
            propsData: {
                mission: configuration.missions[0]
            }
        }).$mount()
    });

    test('snapshot', () => {
        let $html = $mounted.$el.outerHTML;
        expect($html).toMatchSnapshot()
    })
});