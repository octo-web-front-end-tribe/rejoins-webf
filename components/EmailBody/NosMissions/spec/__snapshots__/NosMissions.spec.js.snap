// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NosMissions Component snapshot 1`] = `
"<tr><td dir=\\"ltr\\" bgcolor=\\"#ffffff\\" align=\\"center\\" height=\\"100%\\" valign=\\"top\\" width=\\"100%\\" style=\\"padding: 0px 0px;\\"><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" align=\\"center\\" width=\\"100%\\" style=\\"max-width: 780px;\\"><tr><td align=\\"left\\" valign=\\"top\\" style=\\"font-size: 0; padding: 0px 0px;\\"><div class=\\"stack-column\\" style=\\"display: inline-block; margin: 0px -2px; max-width: 200px; min-width: 160px; vertical-align: top; width: 100%;\\"><table role=\\"presentation\\" cellspacing=\\"0\\" cellpadding=\\"0\\" border=\\"0\\" width=\\"100%\\"><tr><td dir=\\"ltr\\" style=\\"padding: 0px 10px 10px 10px;\\"><div style=\\"text-align: center;\\"><img src=\\"https://s3.amazonaws.com/askbob/users/photos/499/big/CINE5355.jpg?1496785442\\" width=\\"100\\" height=\\"\\" border=\\"0\\" alt=\\"alt_text\\" class=\\"center-on-narrow\\" style=\\"width: 100%; max-width: 100px; height: 100px; margin-top: 20px; margin-bottom: 5px; background: rgb(221, 221, 221); font-family: sans-serif; font-size: 15px; line-height: 140%; color: rgb(85, 85, 85);\\"> <h2 style=\\"margin: 0px; font-family: sans-serif; font-size: 20px; line-height: 125%; color: rgb(20, 160, 192); font-weight: normal;\\">
                                            AJU</h2></div><div style=\\"text-align: center;\\"><img src=\\"https://s3.amazonaws.com/askbob/users/photos/446/big/6957364307006043374-account_id_3.jpg?1506000436\\" width=\\"100\\" height=\\"\\" border=\\"0\\" alt=\\"alt_text\\" class=\\"center-on-narrow\\" style=\\"width: 100%; max-width: 100px; height: 100px; margin-top: 20px; margin-bottom: 5px; background: rgb(221, 221, 221); font-family: sans-serif; font-size: 15px; line-height: 140%; color: rgb(85, 85, 85);\\"> <h2 style=\\"margin: 0px; font-family: sans-serif; font-size: 20px; line-height: 125%; color: rgb(20, 160, 192); font-weight: normal;\\">
                                            OUM</h2></div><div style=\\"text-align: center;\\"><img src=\\"https://s3.amazonaws.com/askbob/users/photos/874/big/IMG_20180205_211241.jpg?1523023940\\" width=\\"100\\" height=\\"\\" border=\\"0\\" alt=\\"alt_text\\" class=\\"center-on-narrow\\" style=\\"width: 100%; max-width: 100px; height: 100px; margin-top: 20px; margin-bottom: 5px; background: rgb(221, 221, 221); font-family: sans-serif; font-size: 15px; line-height: 140%; color: rgb(85, 85, 85);\\"> <h2 style=\\"margin: 0px; font-family: sans-serif; font-size: 20px; line-height: 125%; color: rgb(20, 160, 192); font-weight: normal;\\">
                                            IPA</h2></div></td></tr></table></div> <div class=\\"stack-column\\" style=\\"display: inline-block; margin: 0px -2px; max-width: 70%; min-width: 320px; vertical-align: top;\\"><table role=\\"presentation\\" cellspacing=\\"0\\" cellpadding=\\"0\\" border=\\"0\\" width=\\"100%\\"><tr><td dir=\\"ltr\\" class=\\"center-on-narrow\\" style=\\"font-family: sans-serif; font-size: 15px; line-height: 140%; color: rgb(85, 85, 85); padding: 10px 10px 0px; text-align: left;\\"><h2 style=\\"margin: 0px 0px 10px 0px; font-family: sans-serif; font-size: 24px; line-height: 125%; color: rgb(20, 160, 192); font-weight: bold;\\">
                                        ANEF</h2> <p style=\\"margin: 0px 0px 10px 0px;\\"><div>
    Je suis le fichier de mock pour les markdowns;
</div></p> <br></td></tr></table></div></td></tr> <tr><td><hr style=\\"width: 60%; color: rgb(221, 221, 221);\\"></td></tr></table></td></tr>"
`;
